src/Tools/dotnet-monitor/Profiler/ProfilerService.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Profiler
+{
+    internal sealed class ProfilerService : BackgroundService
+    {
+        private static readonly string SharedLibrarySourcePath = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "shared");
+
+        private readonly TaskCompletionSource<string> _libraryPathSource = new(TaskCreationOptions.RunContinuationsAsynchronously);
+        private readonly ILogger<ProfilerService> _logger;
+        private readonly IOptions<StorageOptions> _storageOptions;
+
+        public ProfilerService(
+            IOptions<StorageOptions> storageOptions,
+            ILogger<ProfilerService> logger)
+        {
+            _logger = logger;
+            _storageOptions = storageOptions;
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            using IDisposable _ = stoppingToken.Register(() => _libraryPathSource.TrySetCanceled(stoppingToken));
+
+            // Yield to allow other hosting services to start
+            await Task.Yield();
```

Comment:
Measure how long the file copy takes in k8s and remove if its negligible.

Reply:
I removed the `Task.Yield` call and used a `StopWatch` to time the initialization call (which basically checks the filesystem and does file copying) to see how long it would take:

- 182 ms
- 278 ms
- 203 ms
- 295 ms
- 190 ms

Average: 230 ms

These measurements were taken on a single Standard_DS2_v2 node that had no other user defined deployments on it.

I'm inclined to keep this `Task.Yield` in place in order to not block startup for this long.